vnext/Microsoft.ReactNative/Views/Image/ReactImage.cpp

```diff
@@ -219,9 +222,15 @@ winrt::fire_and_forget ReactImage::SetBackground(bool fireLoadEndEvent) {
     if (strong_this && fireLoadEndEvent) {
       strong_this->m_onLoadEndEvent(*strong_this, false);
     }
+    co_return;
   }
 
   if (auto strong_this{weak_this.get()}) {
+    // If the image source has been updated since this operation started, do not continue
+    if (currentImageSourceId != strong_this->m_imageSourceId) {
```

Comment:
Note - we will not fire an onLoad / onLoadFailed / onLoadEnd in cases where the image source is updated before the async loading operation completes. I'm not sure what the expectations are in these cases.

Reply:
Yeah, we should check what happens in the core platforms here, since we do want alignment. -- If you are not going to handle that here, at least open an issue to take a look at that later.

Reply:
Done. https://github.com/microsoft/react-native-windows/issues/7194, will update comment.

Reply:
FYI - found source code in iOS that shows similar behavior, documented in #7194 which is closed. There is not currently any implied contract that onLoadEnd fires every time this source changes, so there shouldn't be any concerns here.